app/javascript/layouts/app_router/types/AppRouter__Page.res

```diff
@@ -0,0 +1,75 @@
+exception UnsafeFindFailed(string)
+
+type courseId = string
+type id = string
+
+type t =
+  | Student__Curriculum(courseId)
+  | Student__Report(courseId)
+  | Student__Review(courseId)
+  | Student__Students(courseId)
+  | Student__Leaderboard(courseId)
+  | Student__SubmissionShow(id)
+
+let showSideNav = t => {
+  switch t {
+  | Student__Review(_id)
+  | Student__Curriculum(_id)
+  | Student__Students(_id)
+  | Student__Report(_id)
+  | Student__Leaderboard(_id) => true
+  | Student__SubmissionShow(_id) => false
+  }
+}
+
+let courseId = t =>
+  switch t {
+  | Student__Review(courseId)
+  | Student__Curriculum(courseId)
+  | Student__Students(courseId)
+  | Student__Report(courseId)
+  | Student__Leaderboard(courseId) =>
+    Some(courseId)
+  | Student__SubmissionShow(_id) => None
+  }
+
+let path = t => {
+  switch t {
+  | Student__Curriculum(courseId) => `/courses/${courseId}/curriculum`
+  | Student__Report(courseId) => `/courses/${courseId}/report`
+  | Student__Students(courseId) => `/courses/${courseId}/students`
+  | Student__Review(courseId) => `/courses/${courseId}/review`
+  | Student__Leaderboard(courseId) => `/courses/${courseId}/leaderboard`
+  | Student__SubmissionShow(id) => `/courses/${id}/submission`
```

Comment:
Shouldn't this be...

```suggestion
  | Student__SubmissionShow(submissionId) => `/submissions/${submissionId}/review`
```

I'm pretty certain the route currently held here doesn't exist. 🤔 

There's also possibly a missing test somewhere.